### ISSUE TITLE
before_task_start, after_task_finished don't register for tasks within a namespace

### ISSUE THREAD

**Issue by mbroadhead** [RexOps/Rex#1193]
Rex Version: `(R)?ex 1.6.0`
Rexfile:
```
use Rex -base;
  
package Rex::ACME::SayHello
{
    use Rex -base;
    task say_hello                => sub { say "hello" };
    before_task_start say_hello   => sub { say "before_task_start say_hello" };
    after_task_finished say_hello => sub { say "after_task_finished say_hello" };
}

1;
```

The `before_task_start` and `after_task_finished` hooks do not run, because there was an issue setting them up:
```
$ rex ACME:SayHello:say_hello
[2018-02-21 14:51:40] INFO - Can't add before_task_start say_hello, as it is not yet defined
see /loader/0x25c5fb8/__Rexfile__.pm line 6
[2018-02-21 14:51:40] INFO - Can't add after_task_finished say_hello, as it is not yet defined
see /loader/0x25c5fb8/__Rexfile__.pm line 7
[2018-02-21 14:51:40] INFO - Running task ACME:SayHello:say_hello on <local>
hello
[2018-02-21 14:51:41] INFO - All tasks successful on all hosts
```

If, however, you prefix the hooks with the "rexified package prefix", it works:
Rexfile:
```
use Rex -base;

package Rex::ACME::SayHello
{
    use Rex -base;

    task say_hello => sub {
        say "hello";
    };

    before_task_start 'ACME:SayHello:say_hello' => sub {
        say "before_task_start say_hello";
    };

    after_task_finished 'ACME:SayHello:say_hello' => sub {
        say "after_task_finished say_hello";
    };
}

1;
```

Result:
```
$ rex ACME:SayHello:say_hello
before_task_start say_hello
[2018-02-21 14:56:02] INFO - Running task ACME:SayHello:say_hello on <local>
hello
after_task_finished say_hello
[2018-02-21 14:56:02] INFO - All tasks successful on all hosts
```

My expectation is that `before_task_start`,  `after_task_finished`, and other hooks properly find the task within the same package. Am I missing something here?

**Comment by mbroadhead** [comment]
Here is a quick overview showing that this functionality worked fine on `(R)?ex 1.4.0`, but is broken on `(R)?ex 1.6.0`

Rexfile:
```
use Rex -base;

package Rex::ACME::SayHello {
    use Rex -base;

    task say_hello => sub {
        say "hello";
    };

    before_task_start 'say_hello' => sub {
        say "before_task_start say_hello";
    };

    after_task_finished 'say_hello' => sub {
        say "after_task_finished say_hello";
    };
}

1;
```

```
$ rex -v
(R)?ex 1.4.0

$ rex ACME:SayHello:say_hello  
before_task_start say_hello
[2018-02-22 13:35:06] INFO - Running task ACME:SayHello:say_hello on <local>
hello
after_task_finished say_hello
[2018-02-22 13:35:07] INFO - All tasks successful on all hosts


$ ~/perl5/perlbrew/perls/perl-5.26.1/bin/rex -v
(R)?ex 1.6.0

$ ~/perl5/perlbrew/perls/perl-5.26.1/bin/rex ACME:SayHello:say_hello
[2018-02-22 13:36:05] INFO - Can't add before_task_start say_hello, as it is not yet defined
see /loader/0x22ea3d8/__Rexfile__.pm line 12
[2018-02-22 13:36:05] INFO - Can't add after_task_finished say_hello, as it is not yet defined
see /loader/0x22ea3d8/__Rexfile__.pm line 16
[2018-02-22 13:36:05] INFO - Running task ACME:SayHello:say_hello on <local>
hello
[2018-02-22 13:36:05] INFO - All tasks successful on all hosts

```

**Comment by ferki** [comment]
Thanks for the report, your detailed use-cases, and tests on #1194!

Looks like PR #1164 and PR #1173 were kinda weird, as both were containing each other's whole changeset. So merging one automatically merged the other too. Unfortunately the fix for #1173 was not the correct one if we look at the whole picture, though that fixed its own use-case. You've correctly reverted that change, and that breaks the test that was added for #1173 as expected.

I believe the proper solution is to also move the conversion of the namespace separator out of the `if` block a few lines below, so it always takes place. That seems to make all the tests happy.
